app/dashboard/models.py

```diff
@@ -232,6 +232,7 @@ class Bounty(SuperModel):
     bounty_owner_profile = models.ForeignKey(
         'dashboard.Profile', null=True, on_delete=models.SET_NULL, related_name='bounties_funded', blank=True
     )
+    bounty_reserved_for = JSONField(default=dict, blank=True)
```

Comment:
I'm skeptical that this should be `JSONField`.  Would it be more appropriate to use `reserved_for = models.ForeignKey('dashboard.Profile', null=True, related_name='reserved_bounties', on_delete=models.SET_NULL)` ?

Reply:
@mbeacom  I wanted to use a foreign key relation at first but then realized I'd also need to store the datetime when the issue was reserved for the user for comparison purposes in the UI frontend.

Also when the issue details are modified and the reserved for field is modified I also wanted to capture this as well again for display purposes in the UI i.e. check whether 24 hours are past before allowing others to [start work]

Reply:
so thats why i chose the JSONField 🙂

Reply:
Also finally storing the Profile as the foreign relation, would mean i do another request in the frontend to get the user details  when I want to manipulate them. So I found this to be unnecessary, so thats why I also stored everything for reserved_for in the JSONField and then just get it all at once in the frontend with a single api call.

Reply:
@mul1sh Thanks for the followup!

I understand what you're saying here, my only hang up is if you're storing it as an FK, we'd be able to fetch the reserved data directly in one query, likewise if you're fetching bounty details, you could query once to fetch the profile AND bounty data (just as long as you're using the appropriate `select_related`/`prefetch_related`.  The only time you should have to fetch the reserved user profile independently would be to add it to the bounty initially.

Also, do we care about the datetime when the bounty reserved for data was established? I assume we allow them to reserve for a user profile during bounty creation, so if it's a reserved bounty, we could check against the DT on `Bounty.created_on` and if the delta exceeds 24 hours, drop the reservation and make it open.

Feel free to elaborate if I'm missing something.

Reply:
@mbeacom thanks for clarifying, makes sense now why I should have the foreign key, I actually hadn't thought of using `select_related/prefetch_related` but you are very right with a foreign key this will work.

But still for the bounty I had thought of using `Bounty.created_on` to check if the reserved for time period exceeds 24 hours, but what about when the user updates the issue details and re-assigns the reserved issue to another user? Thats also the reason I chose the JSON object so as to capture this DT. So how do you think I should handle this?

Reply:
Maybe we could use `Activity` since we can already obtain that with a query to `Bounty` via `Bounty.activities`.  We'd have to add a type to `Activity.ACTIVITY_TYPES` and create the relevant object following invited/declined/removed based on the user/funder actions.  I'm not entirely married to this idea, though.

Reply:
Ok thanks for the clarification 🙂 , let me implement this in the next few hours then.

Reply:
@mbeacom one last thing before i push another update, how I get the avatar to display for the reserved_for user? Right now i'm displaying a placeholder but I want to display the users actual avatar in the next commit. Thanks